src/gui/sharedialog.cpp

```diff
@@ -168,6 +169,15 @@ ShareDialog::ShareDialog(AccountPtr account, const QString &sharePath, const QSt
             _ui->checkBox_editing->setEnabled(false);
         }
     }
+
+    /*
+     * Create the share manager and connect it properly
+     */
+    _manager = QSharedPointer<ShareManager>(new ShareManager(_account, this));
```

Comment:
right this goes out of scope immediatly so this looks wrong.
Also, QSharedPointer::create would avoid one alocation.